kroxylicious-filters/kroxylicious-multitenant/src/main/java/io/kroxylicious/proxy/filter/multitenant/MultiTenantTransformationFilterFactory.java

```diff
@@ -6,20 +6,23 @@
 
 package io.kroxylicious.proxy.filter.multitenant;
 
+import java.util.Objects;
+
 import io.kroxylicious.proxy.filter.FilterFactory;
 import io.kroxylicious.proxy.filter.FilterFactoryContext;
+import io.kroxylicious.proxy.filter.multitenant.config.MultiTenantConfig;
 import io.kroxylicious.proxy.plugin.Plugin;
 
-@Plugin(configType = Void.class)
-public class MultiTenantTransformationFilterFactory implements FilterFactory<Void, Void> {
+@Plugin(configType = MultiTenantConfig.class)
+public class MultiTenantTransformationFilterFactory implements FilterFactory<MultiTenantConfig, MultiTenantConfig> {
 
     @Override
-    public Void initialize(FilterFactoryContext context, Void config) {
-        return null;
+    public MultiTenantConfig initialize(FilterFactoryContext context, MultiTenantConfig config) {
+        return config;
     }
 
     @Override
-    public MultiTenantTransformationFilter createFilter(FilterFactoryContext context, Void configuration) {
-        return new MultiTenantTransformationFilter();
+    public MultiTenantTransformationFilter createFilter(FilterFactoryContext context, MultiTenantConfig configuration) {
+        return new MultiTenantTransformationFilter(Objects.requireNonNullElseGet(configuration, () -> new MultiTenantConfig(null)));
```

Comment:
I'm tempted to suggest something like

```suggestion
        return new MultiTenantTransformationFilter(Objects.requireNonNullElse(configuration, DEFAULT_SEPARATOR_CONFIG));
```
Especially as I find the explicit passing of `null` a bit jarring. It does however seem like a reasonable way of centralising the default behaviour.
As that does what we need now but I can see merits in doing what you have.

Reply:
I refactored a bit.  Better?

Reply:
Yes definitely.